src/Streamly/Internal/LZ4/Config.hs

```diff
@@ -25,15 +28,24 @@ where
 -- Imports
 --------------------------------------------------------------------------------
 
+import Data.Coerce (coerce)
 import Data.Int (Int32)
 import Data.Word (Word8)
 import Foreign.Ptr (Ptr, castPtr)
 import Foreign.Storable (peek, poke)
 
+import qualified Streamly.Internal.Data.Array.Storable.Foreign as Array
+
 --------------------------------------------------------------------------------
 -- Configuration
 --------------------------------------------------------------------------------
 
+data ResizeGuard
+    = RGDef -- Run the default computation present
+    | RGAcc -- Accumlate some more array, useful then the ending has a checksum
+    | RGDone -- End successfully
+    | RGFail -- End with a failure, useful to denote a checksum mismatch
+
```

Comment:
I would suggest that we use the data type to identify what type of block are we looking at.

```
data BlockType = DataBlock | FooterPartial | FooterDone | FooterFail
```